bcda/api/requests.go

```diff
@@ -0,0 +1,292 @@
+		// We've added logic into the worker to handle this situation.
+		if err = tx.Commit().Error; err != nil {
+			log.Error(err.Error())
+			oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, responseutils.DbErr, "")
+			responseutils.WriteError(oo, w, http.StatusInternalServerError)
+			return
+		}
+
+		// We've successfully create the job
+		w.Header().Set("Content-Location", fmt.Sprintf("%s://%s/api/v1/jobs/%d", scheme, r.Host, newJob.ID))
+		w.WriteHeader(http.StatusAccepted)
+	}()
+
+	if err = tx.Save(&newJob).Error; err != nil {
+		log.Error(err)
+		oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, responseutils.DbErr, "")
+		responseutils.WriteError(oo, w, http.StatusInternalServerError)
+		return
+	}
+
+	// request a fake patient in order to acquire the bundle's lastUpdated metadata
+	b, err := bb.GetPatient("FAKE_PATIENT", strconv.FormatUint(uint64(newJob.ID), 10), acoID, "", time.Now())
+	if err != nil {
+		log.Error(err)
+		oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, responseutils.FormatErr, "Failure to retrieve transactionTime metadata from FHIR Data Server.")
+		responseutils.WriteError(oo, w, http.StatusInternalServerError)
+		return
+	}
+	newJob.TransactionTime = b.Meta.LastUpdated
+
+	// Decode the _since parameter (if it exists) so it can be persisted in job args
+	var decodedSince string
+	params, ok := r.URL.Query()["_since"]
+	if ok {
+		decodedSince, _ = url.QueryUnescape(params[0])
+	}
+
+	var enqueueJobs []*que.Job
+	enqueueJobs, err = newJob.GetEnqueJobs(resourceTypes, decodedSince, retrieveNewBeneHistData)
+	if err != nil {
+		log.Error(err)
+		oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, responseutils.Processing, "")
+		responseutils.WriteError(oo, w, http.StatusInternalServerError)
+		return
+	}
+	newJob.JobCount = len(enqueueJobs)
+
+	// We've now computed all of the fields necessary to populate a fully defined job
+	if err = tx.Save(&newJob).Error; err != nil {
+		log.Error(err.Error())
+		oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, responseutils.DbErr, "")
+		responseutils.WriteError(oo, w, http.StatusInternalServerError)
+		return
+	}
+
+	// Since we're enqueuing these queuejobs BEFORE we've created the actual job, we may encounter a transient
+	// error where the job does not exist. Since queuejobs are retried, the transient error will be resolved
+	// once we finish inserting the job.
+	for _, j := range enqueueJobs {
+		if err = qc.Enqueue(j); err != nil {
+			log.Error(err)
+			oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, responseutils.Processing, "")
+			responseutils.WriteError(oo, w, http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+func Check429(jobs []models.Job, types []string, w http.ResponseWriter) ([]string, bool) {
```

Comment:
I'm wondering about function visibility here... should this be an exported function?